### ISSUE TITLE
Deterministic condition vs EmbedUntrackedSources

### ISSUE THREAD

**Issue by eddynaka** [NuGetPackageExplorer/NuGetPackageExplorer#1021]
**Type** (choose one): Question
**NPE version**: Latest
**OS version**: Windows 10
**Installed from**: Zip on GitHub

I'm trying to enable a deterministic build using **Azure DevOps** and **MSBuild**. To do that, i added the tags:
```xml
<!-- indicate that the build executes on a build/CI server -->
<ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>

<!-- mapping all source paths included in the project outputs to deterministic values -->
<DeterministicSourcePaths>true</DeterministicSourcePaths>

<!-- produce an assembly whose byte-for-byte output is identical across compilations for identical inputs-->
<Deterministic>true</Deterministic>
```

And, I added this:
```xml
<!-- Embed source files that are not tracked by the source control manager in the PDB -->
<EmbedUntrackedSources>true</EmbedUntrackedSources>
```

When I look at the nupkg generated, it's saying that it's not deterministic. Because i have 3 files that isn't generating the path just like the others:
```
"E:\\A\\_work\\20\\obj\\path\\AssemblyInfo.g.cs"
"E:\\A\\_work\\20\\obj\\path\\...AssemblyAttributes.cs"
"E:\\A\\_work\\20\\obj\\path\\...AssemblyInfo.cs"
```

If I disable the `EmbededUntrackedSources` flag, it removes those, but the SourceLink get's invalid.
How can i achieve both flags to valid?

Thank you.

**Comment by clairernovotny** [comment]
What is the MSBuild version or .NET SDK version you're using for the build? There are fixes in 3.1.300 and 16.6. Prior version have workarounds documented here https://github.com/clairernovotny/DeterministicBuilds.

**Comment by eddynaka** [comment]
hi @clairernovotny , we use this:

Task         : MSBuild
Description  : Build with MSBuild
Version      : 1.166.2
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/build/msbuild

**Comment by clairernovotny** [comment]
What does the build log show? The build agents may not have the latest VS on it.

**Comment by eddynaka** [comment]
Should i look for something specific in the logs?

**Comment by clairernovotny** [comment]
Look at the beginning of the task step and see if it outputs the msbuild version.

**Comment by eddynaka** [comment]
Hi @clairernovotny , found it!

NuGet Version: 5.4.0.6315
MSBuild auto-detection: using msbuild version '16.6.0.22303' from 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\bin'. Use option -MSBuildVersion to force nuget to use a specific version of MSBuild.
MSBuild P2P timeout [ms]: 324000

**Comment by clairernovotny** [comment]
Is the repository public by chance?

@tmat, any idea why the generated files wouldn't have the path map set right?

**Comment by eddynaka** [comment]
hi @clairernovotny , yes it is. It's application insights sdk.

below, the PR that i'm working:
https://github.com/microsoft/ApplicationInsights-dotnet/pull/1907

Right now, I'm trying the workarounds that you suggested and waiting for build.

**Comment by eddynaka** [comment]
ok, i found one of the problem reason: we had this section:
```xml
  <PropertyGroup>
    <!-- Turn on dynamic assembly attribute generation -->
    <AssemblyAttributesPath>$(IntermediateOutputPath)\AssemblyInfo.g.cs</AssemblyAttributesPath>
    <GenerateAdditionalSources>true</GenerateAdditionalSources>
  </PropertyGroup>
```

removing that, one of the problems disapper. but, i still have:
```
<file id="282" name="E:\A\_work\20\obj\Microsoft.ApplicationInsights\Release\src\Microsoft.ApplicationInsights\net45\.NETFramework,Version=v4.5.AssemblyAttributes.cs" language="C#" checksumAlgorithm="SHA256" checksum="xxxx">
<file id="283" name="E:\A\_work\20\obj\Microsoft.ApplicationInsights\Release\src\Microsoft.ApplicationInsights\net45\Microsoft.ApplicationInsights.AssemblyInfo.cs" language="C#" checksumAlgorithm="SHA256" checksum="xxxx">
```

**Comment by eddynaka** [comment]
I made some cleaning in the *.props files but still, couldn't find a way to generate the files correctly.

**Comment by eddynaka** [comment]
@clairernovotny , i tried to debug a little more....my findings: in our project, we have `AssemblyInfo.cs` and some properties in the .csproj/.props. When we have this situation, it generates an unresolved path just like the example:
`
<file id="283" name="E:\A\_work\20\obj\Microsoft.ApplicationInsights\Release\src\Microsoft.ApplicationInsights\net45\Microsoft.ApplicationInsights.AssemblyInfo.cs" language="C#" checksumAlgorithm="SHA256" checksum="xxxx">` 

So, what I did:
- removed everything from the .csproj / .props, added the property to don't generate the assembly info and added all the content to AssemblyInfo.cs. With that, that file doesn't appear in the xml inside the nupkg.

For the other file that appears in the xml: `<file id="282" name="E:\A\_work\20\obj\Microsoft.ApplicationInsights\Release\src\Microsoft.ApplicationInsights\net45\.NETFramework,Version=v4.5.AssemblyAttributes.cs" language="C#" checksumAlgorithm="SHA256" checksum="xxxx">` even adding the tags that you proposed in your example. That didn't work for me.

I tried to download your example and simple pack with dotnet core (using net5 preview4) or msbuild, both doesn't generate deterministic / source link available
